File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    bracket = list(s)
    for i in bracket:
        if i == "(":
            stack.append("(")
        else:
            if len(stack) == 0 :
                return False
            else:
                stack.pop()
    if len(stack) != 0:
        return False
    
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        
        char[] arrays = s.toCharArray();
        
        boolean answer = solve(arrays);
        
        return answer;
    }
    
    static void print(Object s) {
        System.out.println(s);
    }
    
    static boolean solve(char[] arrays) {
        
        Stack<Character> stack = new Stack<Character>();
        
        for(char c : arrays) {
            if (c == '(') {
                stack.push(c);
            }
            
            if (c == ')') {
                if (stack.size() > 0) {
                    stack.pop();
                }
                else {
                    return false;
                }
            }
        }
        
        return !(stack.size() > 0);
    }
}